### PR TITLE
add next-firebase-auth package to journeys-admin

### PR DESCRIPTION
**Pull request by tataihono**
# DEV NOTES
When this is merged you'll need to update your .env.local file with:
- GOOGLE_APPLICATION_JSON (copy from api-gateway project)
- COOKIE_SECRET_CURRENT (set to a random string of characters)

## api-journeys changes
- allow userJourneyRequest mutation to be called with a slug and a database id

@mikeallisonJS contributed the following API changes
- add adminJourney query that returns a journey from journeys the current user has access to
- add adminJourneys query that returns journeys the current user has access to
- change journeys to only return published journeys
- change journey query to only return a journey from published journeys

## journeys-admin changes
- remove new journey creation
- route /journeys to /
- use next-firebase-auth for authentication
	- add /api/login
	- add /api/logout
	- update getServerSideProps of all pages to handle firebase tokens and provider
- add apollo cache hydration
	- apollo client initialization to receive firebase token
	- update getServerSideProps to store successful graphql calls in cache
	- run the query on client-side and draw results from the cache
- add sign-in page and handle automatic routing with support for redirection after sign-in
- remove invite page in favour of just using the single journey page when unauthorized
- add journeyInvite component
- @storyworks remove button margin

### CODE REVIEW

File: apps/api-journeys/src/app/modules/userJourney/userJourney.resolver.ts [hunk 1]
```diff
@@ -36,11 +37,19 @@ export class UserJourneyResolver {
...
+      idType === IdType.slug
+        ? await this.journeyService.getBySlug(journeyId)
+        : await this.journeyService.get(journeyId)
+
+    if (journey == null) throw new UserInputError('journey does not exist')
+
     return await this.userJourneyService.save({
       userId,
-      journeyId,
+      journeyId: (journey as { _key: string })._key,
```
Comment (mikeallisonJS) [inline apps/api-journeys/src/app/modules/userJourney/userJourney.resolver.ts:52]: not necessary, but can be shortened using lodash get(journey, '_key')

**Review by GeronimoJohn — APPROVED** [review]

**Review by GeronimoJohn — DISMISSED** [review]

File: apps/api-journeys/src/app/modules/userJourney/userJourney.resolver.ts (hunk 1 above)
Comment (tataihono) [reply apps/api-journeys/src/app/modules/userJourney/userJourney.resolver.ts:52]: will leave as is. should really fix underlying types in future.

**Comment by tataihono** [comment]
@GeronimoJohn and @mikeallisonJS: this PR is ready to be reviewed again. Please also update your .env.local file for journeys-admin.

File: apps/journeys-admin/pages/index.tsx [hunk 2]
```diff
@@ -1,45 +1,80 @@
...
-      <Typography
-        variant={'body2'}
-        sx={{ mt: 20, color: 'primary.main', cursor: 'pointer' }}
-        onClick={handleFeedbackSupport}
-      >
-        Feedback & Support
-      </Typography>
-    </Box>
+    <>
+      <JourneysAppBar variant="list" />
```
Comment (storyworks) [inline apps/journeys-admin/pages/index.tsx:52]: We removed this from the page since it was already added to JourneyList (so it could be included in VR)

File: apps/journeys-admin/src/components/JourneyList/JourneyList.tsx [hunk 3]
```diff
@@ -27,14 +26,12 @@ export function JourneyList({ journeys }: JourneysListProps): ReactElement {
...
     <>
-      <JourneysAppBar variant={'list'} />
       <Stack
         direction={{ xs: 'column', sm: 'row' }}
         spacing={2}
         sx={{
           justifyContent: 'space-between',
-          m: 6,
-          mt: 20
+          m: 6
```
Comment (storyworks) [inline apps/journeys-admin/src/components/JourneyList/JourneyList.tsx:34]: Revert the changes in this file?

File: apps/journeys-admin/src/components/JourneysAppBar/JourneysAppBar.tsx [hunk 4]
```diff
@@ -54,6 +54,7 @@ export function JourneysAppBar({ variant }: JourneysAppBarProps): ReactElement {
           </Toolbar>
         )}
       </AppBar>
+      <Toolbar />
```
Comment (storyworks) [inline apps/journeys-admin/src/components/JourneysAppBar/JourneysAppBar.tsx:57]: This toolbar was being used to insert a random gap and divider underneath the actual toolbar on desktop. Remove..

File: apps/journeys-admin/src/components/JourneyView/JourneyView.tsx [hunk 5]
```diff
@@ -17,7 +17,7 @@ export function JourneyView(): ReactElement {
   return (
     <Box sx={{ mr: breakpoints.md ? '328px' : 0 }}>
       <JourneysAppBar variant="view" />
-      <Box sx={{ p: 8, pt: 20, backgroundColor: 'background.paper' }}>
+      <Box sx={{ p: 8, backgroundColor: 'background.paper' }}>
```
Comment (storyworks) [inline apps/journeys-admin/src/components/JourneyView/JourneyView.tsx:20]: Revert change? Needed to add some top padding after removing the space caused by a duplicate toolbar.

**Review by storyworks — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/JourneyView/JourneyView.tsx (hunk 5 above)
Comment (tataihono) [reply apps/journeys-admin/src/components/JourneyView/JourneyView.tsx:20]: Future PR incoming.

File: apps/journeys-admin/src/components/JourneyList/JourneyList.tsx (hunk 3 above)
Comment (tataihono) [reply apps/journeys-admin/src/components/JourneyList/JourneyList.tsx:34]: Future PR incoming.

File: apps/journeys-admin/pages/index.tsx (hunk 2 above)
Comment (tataihono) [reply apps/journeys-admin/pages/index.tsx:52]: Future PR incoming.

File: apps/journeys-admin/src/components/SignIn/SignIn.tsx [hunk 6]
```diff
@@ -1,58 +1,81 @@
...
+        <Box
+          sx={{
+            display: 'flex',
+            flexDirection: 'column',
+            alignItems: 'center',
+            justifyContent: 'center',
+            pt: 30
+          }}
+        >
+          <Image src={`/${logo}`} alt="Next Steps" height={68} width={152} />
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/SignIn/SignIn.tsx:61]: Getting an error with the logo - see photo below. 

I tested changing the syntax to be `src={logo}` and that seemed to fix the error below

Screen Shot 2022-01-21 at 9 15 13 AM

**Review by mikeallisonJS — DISMISSED** [review]
api loooks good

File: apps/journeys-admin/src/components/JourneyInvite/JourneyInvite.tsx [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+          display: 'flex',
+          flexDirection: 'column',
+          alignItems: 'center',
+          justifyContent: 'center',
+          pb: 10
+        }}
+      >
+        <Link href="/" passHref>
+          <a>
+            <Image src={`/${logo}`} alt="Next Steps" height={68} width={152} />
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/JourneyInvite/JourneyInvite.tsx:57]: The image is not loading for me here as well

Comment (tataihono) [reply apps/journeys-admin/src/components/JourneyInvite/JourneyInvite.tsx]: fixed

**Review by GeronimoJohn — APPROVED** [review]
Front-end changes looks good!
